universum/modules/code_report_collector.py

```diff
@@ -65,17 +65,17 @@ def _process_one_sarif_issue(self, issue, root_uri_base_paths, who) -> None:
                 if location_data.get('address'):
                     continue  # binary artifact can't be processed
                 raise ValueError("Unexpected lack of artifactLocation tag")
+            uri = artifact_data.get('uri')
```

Comment:
1. Wouldn't `if not uri` check also cover the case of empty `uri` element?
2. Per my understanding of SARIF spec, the only option when `uri` may be absent in `artifactLocation` is when there is an array of artifacts in the document, and `artifactLocation` has `index` element instead. As Universum doesn't currently support this use case, I opted for an explicit error. If this is not desired, I can revert the behavior to using empty `path` instead

Reply:
I am talking about missing `uri` element, not empty one. The second parameter of the `get` method of the dict ensures that some value is returned. Otherwise it throws an error. So the only potential issue would be that the error is harder to understand than with the previous implementation.
I think it would be enough to just manually check this case.

Reply:
Ok, correction: the `get` method doesn't throw an error. It returns `None` instead. So the code is correct.